plugin/metrics/prometheus/metricsstore/reader.go

```diff
@@ -287,17 +288,20 @@ func promqlDurationString(d *time.Duration) string {
 	return string(b)
 }
 
-func startSpanForQuery(ctx context.Context, metricName, query string) (opentracing.Span, context.Context) {
-	span, ctx := opentracing.StartSpanFromContext(ctx, metricName)
-	ottag.DBStatement.Set(span, query)
-	ottag.DBType.Set(span, "prometheus")
-	ottag.Component.Set(span, "promql")
-	return span, ctx
+func startSpanForQuery(ctx context.Context, metricName, query string) (context.Context, trace.Span) {
+	tp := jtracer.New().OTEL
```

Comment:
We never init tracers in lower level components, only in main.

Reply:
Sorry @yurishkuro , any way to initialise it. I mean `package main` ?

Reply:
you already added code to initialize it in main. You just need to pass it down to components, similar to other dependencies like metrics factory or logger.